### DISCUSSION TITLE
Improved Face Detection

### DISCUSSION THREAD

**Discussion by hjed** [photoprism/photoprism#2277] [Q&A] [answered]
Hi there

I'm in the process of moving over from Google Photos - however one of the things I'm running into is that photoprism detects far less faces in photos (at first I though it was grouping that wasn't working as well as I liked but it looks like the faces aren't being detected in the first place). 

I was wondering if there where any plans to improve this (I've tried playing around with the face size settings but doesn't seem to make much difference). Or alternatively I'm a developer (although a time poor one) if there are simple ways that this can be done (without going into the guts of the ML algos you are using which I probably don't have time for) I'd potentially be interested in helping. 

Here's an example of the type of photos I'm having issues with: 
![PXL_20220418_013718895](https://user-images.githubusercontent.com/6956262/164966193-77414f55-e04b-42cd-85b2-7da6833b235a.jpg)

There are four faces in this photo that I'd expect a computer to be able to detect. Currently photoprism only detects one (the one on the right facing dead on to the camera) and doesn't group it. However with the same photo (and the same set of other photos for these particular people) Google detects 3 faces and groups two of them (the three facing the camera except the far left one). From the random sampling I've done this seems to be fairly consistent. 

I recognise that obviously the grouping, etc is not going to be as powerful as Google given their large datasets. But I'd really like to see it much closer. 

Thanks for any help.

Cheers,
Harry

**Comment by graciousgrey** [comment, marked as answer]
Many improvements for facial recognition (such as improved detection, manual tagging etc) are planned.

**Reply by hjed** [reply]
Awesome, good to hear!

**Comment by ndfan77** [comment]
IMHO (and as a paid Plus subscriber), the lack of robust face recognition is (by far) the biggest current weakness in Photo Prism today.  I'm actually considering leveraging Tonfotos for face recognition and then throwing together a Python script that takes it's results (from .ini files saved in each directory) and add them to the Xmp.iptcExt.PersonInImage tag -- because PP's face detection is so weak.

Why not at least offer some variable change suggestions to the [Face Recognition documentation](https://docs.photoprism.app/getting-started/config-options/#face-recognition) that might produce better/more aggressive face detection (e.g. --face-size 20 --face-score 4 --thumb-size 4096, etc...)?  

It seems like it would be better to err on the side of more faces detected with more errors (which can be fixed), than to err on the side of less faces detected with fewer errors (which isn't as easy to correct).

**Reply by gombihu** [reply]
Second to this topic. I don't get it why for obvious pictures it does not make face recognition.

**Comment by luposlip** [comment]
Ref.: https://github.com/photoprism/photoprism/discussions/4028
